src/ui/public/agg_types/buckets/geo_hash.js

```diff
@@ -1,34 +1,34 @@
 import _ from 'lodash';
-import moment from 'moment';
 import AggTypesBucketsBucketAggTypeProvider from 'ui/agg_types/buckets/_bucket_agg_type';
 import precisionTemplate from 'ui/agg_types/controls/precision.html';
+import {geohashCells} from 'ui/utils/decode_geo_hash';
+
 export default function GeoHashAggDefinition(Private, config) {
   let BucketAggType = Private(AggTypesBucketsBucketAggTypeProvider);
   let defaultPrecision = 2;
 
-  // zoomPrecision maps event.zoom to a geohash precision value
-  // event.limit is the configurable max geohash precision
-  // default max precision is 7, configurable up to 12
-  const zoomPrecision = {
-    1: 2,
-    2: 2,
-    3: 2,
-    4: 3,
-    5: 3,
-    6: 4,
-    7: 4,
-    8: 5,
-    9: 5,
-    10: 6,
-    11: 6,
-    12: 7,
-    13: 7,
-    14: 8,
-    15: 9,
-    16: 10,
-    17: 11,
-    18: 12
-  };
+
+  /**
+   * Map Leaflet zoom levels to geohash precision levels.
+   * The size of a geohash grid on the map should be at least `mininumGeohashSizeInPixels` pixels wide.
+   */
+  let zoomPrecision = {};
+  const minGeohashPixels = 16;
+  for (let zoom = 0; zoom <= 18; zoom += 1) {
```

Comment:
Kept it 18 because that was the limit of the previous one. Indeed, moving to 21 will be more robust.